pkg/svc/zserver.go

```diff
@@ -2,51 +2,89 @@ package svc
 
 import (
 	"context"
+	"strconv"
+	"time"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/my-responder/pkg/dapr"
 	"github.com/my-responder/pkg/pb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~ A BRIEF DEVELOPMENT GUIDE ~~~~~~~~~~~~~~~~~~~~~~~~~
-//
-// TODO: Extend the MyResponder service by defining new RPCs and
-// and message types in the pb/service.proto file. These RPCs and messages
-// compose the API for your service. After modifying the proto schema in
-// pb/service.proto, call "make protobuf" to regenerate the protobuf files.
-//
-// TODO: Create an implementation of the MyResponder server
-// interface. This interface is generated by the protobuf compiler and exists
-// inside the pb/service.pb.go file. The "server" struct already provides an
-// implementation of MyResponder server interface service, but only
-// for the GetVersion function. You will need to implement any new RPCs you
-// add to your protobuf schema.
-//
-// TODO: Update the GetVersion function when newer versions of your service
-// become available. Feel free to change GetVersion to better-suit how your
-// versioning system, or get rid of it entirely. GetVersion helps make up a
-// simple "starter" example that allows an end-to-end example. It is not
-// required.
-//
-// TODO: Oh yeah, delete this guide when you no longer need it.
-//
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~ FAREWELL AND GOOD LUCK ~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
 const (
 	// version is the current version of the service
 	version = "0.0.1"
 )
 
 // Default implementation of the MyResponder server interface
-type server struct{}
+type server struct {
+	pb.UnimplementedMyResponderServer
+	Description string
+	Timestamp   time.Time
+	Requests    int64
+	pubsub      *dapr.PubSub
+}
 
 // GetVersion returns the current version of the service
 func (server) GetVersion(context.Context, *empty.Empty) (*pb.VersionResponse, error) {
 	return &pb.VersionResponse{Version: version}, nil
 }
 
+func (s *server) GetDescription(ctx context.Context, req *pb.GetDescriptionRequest) (*pb.GetDescriptionResponse, error) {
+	if req.GetService() != 2 {
+		data := []byte("ping")
+		s.pubsub.Publish("info", data)
+		return &pb.GetDescriptionResponse{Description: <-s.pubsub.Buffer}, nil
```

Comment:
Are you sure you will receive a description from channel? Pubsub does not guarantee that few requests, published into publish topic, will be processed on remote side and send back in the proper order, or will be received at all.

Reply:
Fixed